src/config/authd-config.c

```diff
@@ -231,10 +232,13 @@ int Read_Authd(const OS_XML *xml, XML_NODE node, void *d1, __attribute__((unused
 
         config->force_options.enabled = legacy_force_insert;
     }
-    if (legacy_force_time > 0) {
-        mdebug1("Setting <force><enabled> tag to %s to comply with the legacy <%s> option found.",
-                legacy_force_time ? "'yes'" : "'no'", xml_force_time);
-        config->force_options.enabled = legacy_force_time;
+    if (legacy_force_time != -1) {
+        mdebug1("Setting <force>< disconnected_time> tag to %d to comply with the legacy <%s> option found.",
+                legacy_force_time, xml_force_time);
+        if(legacy_force_time != 0) {
+            config->force_options.enabled = false;
```

Comment:
The setting that should be modified is `config->force_options.disconnected_time_enabled`.

In addition, the logic is not correct. `legacy_force_time == 0` means that we always want to replace the agents, no matters the time they have been disconnected, so `config->force_options.disconnected_time_enabled` is `false` when `legacy_force_time` is 0.

Please, apply this block:
```
if (legacy_force_time == 0) {
    config->force_options.disconnected_time_enabled = false;
} else {
    config->force_options.disconnected_time_enabled = true;
    config->force_options.disconnected_time = legacy_force_time;
}
```